File: src/test/java/sirius/db/text/TokenProcessorTest.java
/*
 * Made with all the love in the world
 * by scireum in Remshalden, Germany
 *
 * Copyright by scireum GmbH
 * http://www.scireum.de - info@scireum.de
 */

package sirius.db.text;

import org.junit.Assert;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.List;

/**
 * Provides baseclass used to test <tt>TokenProcessors</tt>.
 */
abstract class TokenProcessorTest {

    public static final String[] NO_TOKENS = new String[0];

    protected static void assertExactTokenizing(String input, ChainableTokenProcessor processor, String... tokens) {
        assertExactTokenizing(Collections.singletonList(input), processor, tokens);
    }

    protected static void assertExactTokenizing(List<String> inputs,
                                                ChainableTokenProcessor processor,
                                                String... tokens) {
        List<String> result = new ArrayList<>();
        processor.chainConsumer(result::add);
        inputs.forEach(processor);
        processor.purge();
        Assert.assertEquals(Arrays.asList(tokens), result);
    }
}

File: src/test/kotlin/sirius/db/text/TokenProcessorTest.kt
/*
 * Made with all the love in the world
 * by scireum in Remshalden, Germany
 *
 * Copyright by scireum GmbH
 * http://www.scireum.de - info@scireum.de
 */

package sirius.db.text

import kotlin.test.assertEquals

/**
 * Provides baseclass used to test <tt>TokenProcessors</tt>.
 */
abstract class TokenProcessorTest {
    companion object {
        val NO_TOKENS: Array<String?> = arrayOfNulls(0)
    }

    protected fun assertExactTokenizing(
        input: String,
        processor: ChainableTokenProcessor,
        vararg tokens: String?
    ) {
        val result: MutableList<String> = ArrayList()
        processor.chainConsumer { token: String -> result.add(token) }
        listOf(input).forEach(processor)
        processor.purge()
        assertEquals(listOf(*tokens), result)
    }
}
